esp-hal/src/trace.rs

```diff
@@ -57,22 +53,15 @@ pub struct TraceResult {
 }
 
 /// TRACE Encoder Instance
-pub struct Trace<'d, T>
-where
-    T: Instance,
-{
-    peripheral: PeripheralRef<'d, T>,
+pub struct Trace<'d> {
+    peripheral: TRACE0<'d>,
```

Comment:
certainly fine for now but we need to make it generic again for P4

Reply:
Can't we have AnyTrace?